app/javascript/packs/articlePage.jsx

```diff
@@ -47,17 +47,13 @@ if (shareDropdownButton.dataset.initialized !== 'true') {
     // We want to close the dropdown on link select (since they open in a new tab)
     document
       .querySelectorAll('#article-show-more-dropdown [href]')
-      .forEach((link) => link.addEventListener('click', closeDropdown));
-
-    // Add Ahoy stats for dropdown share options
-    const container = document.getElementsByClassName(
-      'crayons-article-actions__inner',
-    )[0];
-    container?.querySelectorAll('a.dropdown-link-row').forEach((link) => {
-      link.addEventListener('click', (event) => {
-        ahoy.track('Post Dropdown', { option: event.target.text.trim() });
+      .forEach((link) => {
+        link.addEventListener('click', closeDropdown);
+        // Temporary Ahoy Stats for usage reports
+        link.addEventListener('click', (event) => {
+          ahoy.track('Post Dropdown', { option: event.target.text.trim() });
+        });
```

Comment:
Refactor (blocking): Two click event listeners is not necessary. Merge the body of each of these event listener callbacks into one callback.

```suggestion
        link.addEventListener('click', (event) => {
          closeDropdown(event)
          
          // Temporary Ahoy Stats for usage reports
          ahoy.track('Post Dropdown', { option: event.target.text.trim() });
        });
```

or add the ahoy tracking directly in `closeDropdown`.

Reply:
Thanks @nickytonline, this is indeed a much better way to do it!